### PR TITLE
Fix issue 258 265

### PR DESCRIPTION
**Pull request by dadhi**
Related to #258 and #265

### CODE REVIEW

**Comment by Havunen** [comment]
@dadhi what is the status with this PR? Does it still need some work or is it just waiting to get merged?

I started some test runs yesterday evening but it was from commit

"unsplit Func<object> invocation interpretation"

I will check the results when I'm home later today

**Comment by dadhi** [comment]
Current status:

- FEC is working with the hardcoded split to >=256 split sub-graph
- WithoutFEC does not work yet, currently WIP

**Comment by Havunen** [comment]
Hi,

I see I just arrived home and WithoutFEC is failing depth > 10 to stackoverflow exception.

**Comment by dadhi** [comment]
Depth split would be removed anyway, I am testing other approach based on total number of dependencies split.
I am not sure what to do with the option, likely to mark it as Obsolete pointing to the replacement option.

**Comment by Havunen** [comment]
Okay. I wonder what would be the best way to verify its working correctly to avoid surprises in production? Maybe control this split treshold in test case and loop through load test using different numbers? 

Or... Have some code generator to make insane depth levels so different use cases could be tested.. but maintaining this might be difficult

File: src/DryIoc/Container.cs [hunk 1]
```diff
@@ -9561,22 +9652,63 @@ public virtual Expression GetExpressionOrDefault(Request request)
...
+                {
+                    // it does not make sense to split the scoped or singleton graphs
+                    // because former already includes the lambda and latter is a constant or lambda
+                    if (reuse == DryIoc.Reuse.Transient &&
+                        !rules.UsedForValidation && !rules.UsedForExpressionGeneration) 
+                    {
+                        // Split the expression with dependencies bigger than certain threshold by wrapping it in Func which is a
+                        // separate compilation unit and invoking it emmediately
+                        var depCount = request.DependencyCount;
+                        if (depCount >= 1024)
```
Comment (Havunen) [inline src/DryIoc/Container.cs:9670]: @dadhi What is this number 1024 based on? Is there any reason why its better than 128? or any other number? This to me looks some kind of treshold variable, so should this be configured so we can test it looping through different values?

**Comment by dadhi** [comment]
> What is this number 1024 based on

I am testing how many lambdas the number produces in ReducedLoadTest. I want to find something that just works. 

Allowing to change it is strange, because the person must know a lot of DryIoc implementation details. 

Also testing the number

**Comment by Havunen** [comment]
> Also the question, what will you do with the results of testing the numbers? Big number may produce SO exception, but that's exactly why we are making it smaller.

I just want to verify the code works in possible edge cases introduced. Having 1024 depcount could lead into another similar bug where adding one dependency produces invalid program exception. when dependency count grows to 1025. Idea of having it configurable is for testing only, so if any bug is found then unit test can be written without 1025 dependencies

**Comment by dadhi** [comment]
Hmm Ok, I fine with that. Let's make it configurable and see if it make sense.

**Comment by Havunen** [comment]
How much is still to be done here? We are eagerly waiting for these bugs to get fixed :)
4.2.0 hype

**Comment by dadhi** [comment]
I also want to finish it asap being tired of it.

**Pull request merged**